lambdas/services/process_mns_message_service.py

```diff
@@ -0,0 +1,150 @@
+import os
+from datetime import datetime
+
+from botocore.exceptions import ClientError
+from enums.death_notification_status import DeathNotificationStatus
+from enums.metadata_field_names import DocumentReferenceMetadataFields
+from enums.mns_notification_types import MNSNotificationTypes
+from enums.patient_ods_inactive_status import PatientOdsInactiveStatus
+from models.mns_sqs_message import MNSSQSMessage
+from services.base.dynamo_service import DynamoDBService
+from services.base.sqs_service import SQSService
+from utils.audit_logging_setup import LoggingService
+from utils.exceptions import PdsErrorException
+from utils.utilities import get_pds_service
+
+logger = LoggingService(__name__)
+
+
+class MNSNotificationService:
+    def __init__(self):
+        self.dynamo_service = DynamoDBService()
+        self.table = os.getenv("LLOYD_GEORGE_DYNAMODB_NAME")
+        self.pds_service = get_pds_service()
+        self.sqs_service = SQSService()
+        self.queue = os.getenv("MNS_NOTIFICATION_QUEUE_URL")
+
+    def handle_mns_notification(self, message: MNSSQSMessage):
+        try:
+            if message.type == MNSNotificationTypes.CHANGE_OF_GP:
+                logger.info("Handling GP change notification.")
+                self.handle_gp_change_notification(message)
+
+            elif message.type == MNSNotificationTypes.DEATH_NOTIFICATION:
+                logger.info("Handling death status notification.")
+                self.handle_death_notification(message)
+
+        except PdsErrorException:
+            logger.info("An error occurred when calling PDS")
+            self.send_message_back_to_queue(message)
+
+        except ClientError as e:
+            logger.info(
+                f"Unable to process message: {message.id}, of type: {message.type}"
+            )
+            logger.info(f"{e}")
+
+    def handle_gp_change_notification(self, message: MNSSQSMessage):
+        patient_document_references = self.get_patient_documents(
+            message.subject.nhs_number
+        )
+
+        if not self.patient_is_present_in_ndr(patient_document_references):
+            return
```

Comment:
Given that both the change of GP and the death notifications (and I'd imagine any potential future notifications) involve this step of pulling out the document references and then checking for the presence of this patient in the NDR, I'd suggest we pull these lines out and do this check on line 29 before the `if/elif` to avoid duplication and clean these steps up.